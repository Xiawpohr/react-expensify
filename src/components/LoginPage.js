import React from 'react'
import { connect } from 'react-redux'
import { login } from '../actions/auth'

export const LoginPage = (props) => (
  <div>
    <button onClick={props.login} >Login</button>
  </div>
)

const mapDispatchToProps = (dispatch) => ({
  login: () => dispatch(login())
})

export default connect(undefined, mapDispatchToProps)(LoginPage)
