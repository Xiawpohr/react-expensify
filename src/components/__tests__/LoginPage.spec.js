import React from 'react'
import { shallow } from 'enzyme'
import { LoginPage } from '../LoginPage.js'

let login, wrapper

beforeEach(() => {
  login = jest.fn()
  wrapper = shallow(<LoginPage login={login} />)
})

test('should render Login Page correctly', () => {
  expect(wrapper).toMatchSnapshot()
})

test('should handle login on click evenet', () => {
  wrapper.find('button').simulate('click')
  expect(login).toHaveBeenCalledTimes(1)
})
