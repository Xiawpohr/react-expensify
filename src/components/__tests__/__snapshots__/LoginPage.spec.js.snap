// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Login Page correctly 1`] = `
<div>
  <ForwardRef
    url="bg.jpg"
  >
    <ForwardRef>
      <ForwardRef>
        Expensify
      </ForwardRef>
      <p>
        It's time to get your expenses under control.
      </p>
      <ForwardRef
        onClick={[Function]}
      >
        Login With Google
      </ForwardRef>
    </ForwardRef>
  </ForwardRef>
</div>
`;
